### PR TITLE
Updates for latest rust

### PR DESCRIPTION
**Pull request by richo**
I've started trying to get Zinc working on modern rust.

I'm currently blocked on https://github.com/rust-lang/rust/pull/32269 but this at least massively cuts down on the compile errors.

### CODE REVIEW

File: examples/blink_k20/src/main.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ pub fn main() {
   watchdog::init(watchdog::State::Disabled);
 
   // Pins for MC HCK (http://www.mchck.org/)
-  let led1 = pin::Pin::new(pin::Port::PortB, 16, pin::Function::Gpio, Some(zinc::hal::pin::Out));
+  let led1 = pin::Pin::new(pin::Port::PortB, 13, pin::Function::Gpio, Some(zinc::hal::pin::Out));
```
Comment (richo) [inline examples/blink_k20/src/main.rs]: Oops, I'll fix this

**Comment by richo** [comment]
This gets zing building cleanly.

I will hopefully have a look at the tests tomorrow.

**Comment by farcaller** [comment]
Looks nice, thanks for your work. I'll keep this open until we get green on travis though.

**Comment by richo** [comment]
Works for me! There's some pretty severe breakage in the way that ASTs are constructed, and my lack of familiarity with the code is making debugging hard.

If it's ok with you I'll braindump into this issue and see if anything makes more sense to you?

**Comment by farcaller** [comment]
Sounds good.

**Comment by posborne** [comment]
As an aside, we've been using homu (http://homu.io/) on nix and it has worked out pretty well.  https://github.com/nix-rust/nix/blob/master/CONTRIBUTING.md#homu-the-bot-who-merges-all-the-prs.  Homu will verify that the resultant merged state passes before merging into master.

**Comment by farcaller** [comment]
Cool, thanks, I'll take a look at it (it's the next iteration of bors, right?)

**Comment by posborne** [comment]
Yes, I'm not sure if it is the next generation or just the generally available version.  Either way, yes -- it's basically bors.

**Comment by Gyscos** [comment]
This fixes `ioreg`, but it seems `platformtree` still uses some of the old types (for instance `platformtree/src/parser.rs:19` imports `syntax::ast::LitInt`).

**Comment by richo** [comment]
Thanks to @Gyscos I was able to build blinky_lpc17xx using rust 1.9-dev.

I'm curious to see what travis will make of the tests, but this is progressing.

**Comment by Gyscos** [comment]
#374, already merged, includes most of the fixes present here.

**Comment by mcoffin** [comment]
> already merged

Yep. Closing this.

**Pull request closed without merging**